Handling of Left-to-Right Mark (U+200E) in URLs

## Summary

We are evaluating the Granian WSGI server for use with our Django application. During this testing phase, we have encountered an issue with how Granian handles certain characters, specifically the Left-to-Right Mark (U+200E), when present in URLs.

## Steps to Reproduce

1. Run a Django application using the built-in development server: `python manage.py runserver`
2. Make a request to a URL containing the Left-to-Right Mark (U+200E) character, e.g., `curl http://localhost:8000/%E2%80%8E`

With the built-in Django development server, the request is processed correctly, and the expected response is returned.

However, when running the same Django application with Granian as the WSGI server:

1. Start the Granian server: `granian wsgi.app:application --interface wsgi`
2. Make the same request to the URL containing the Left-to-Right Mark (U+200E) character: `curl http://localhost:8000/%E2%80%8E`

In this case, Granian triggers an error, failing to process the request correctly.

## Expected Behavior

The expected behavior is for Granian to handle URLs containing the Left-to-Right Mark (U+200E) character in the same way as Django's built-in development server and other WSGI servers like Gunicorn and uWSGI.

## Additional Context

- Other WSGI servers like Gunicorn and uWSGI handle URLs containing the Left-to-Right Mark (U+200E) character correctly when used with the same Django application.
- This issue has been observed with the latest version of Granian (v1.4.2) at the time of this report.

This appears to be a difference in the type of string passed down to Django. There is a code comment above this line:
```
    # Non-ASCII values in the WSGI environ are arbitrarily decoded with
    # ISO-8859-1. This is wrong for Django websites where UTF-8 is the default.
    # Re-encode to recover the original bytestring.
```
https://github.com/django/django/blob/e2428292abaca4758a7508175d31667fe2dff57c/django/core/handlers/wsgi.py#L197

## Traceback

```traceback
[ERROR] Application callable raised an exception
Traceback (most recent call last):
  File "/Users/rob/.pyenv/versions/3.11.8/envs/bedrock/lib/python3.11/site-packages/granian/wsgi.py", line 40, in _runner
    rv = callback(scope, resp)
         ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/git/bedrock/wsgi/app.py", line 15, in https_application
    return django_application(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/.pyenv/versions/3.11.8/envs/bedrock/lib/python3.11/site-packages/django/core/handlers/wsgi.py", line 123, in __call__
    request = self.request_class(environ)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/.pyenv/versions/3.11.8/envs/bedrock/lib/python3.11/site-packages/django/core/handlers/wsgi.py", line 61, in __init__
    path_info = get_path_info(environ) or "/"
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/.pyenv/versions/3.11.8/envs/bedrock/lib/python3.11/site-packages/django/core/handlers/wsgi.py", line 149, in get_path_info
    path_info = get_bytes_from_wsgi(environ, "PATH_INFO", "/")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rob/.pyenv/versions/3.11.8/envs/bedrock/lib/python3.11/site-packages/django/core/handlers/wsgi.py", line 197, in get_bytes_from_wsgi
    return value.encode("iso-8859-1")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'latin-1' codec can't encode character '\u200e' in position 9: ordinal not in range(256)
```

--- a/src/wsgi/callbacks.rs
+++ b/src/wsgi/callbacks.rs
@@ -11,7 +11,7 @@ use itertools::Itertools;
 use percent_encoding::percent_decode_str;
 use pyo3::{
     prelude::*,
-    types::{IntoPyDict, PyDict},
+    types::{IntoPyDict, PyBytes, PyDict},
 };
 use std::borrow::Cow;
 use std::net::SocketAddr;
@@ -39,7 +39,7 @@ fn run_callback(
         .uri
         .path_and_query()
         .map_or_else(|| ("", ""), |pq| (pq.path(), pq.query().unwrap_or("")));
-    let path = percent_decode_str(path_raw).decode_utf8().unwrap();
+    let path = percent_decode_str(path_raw).collect_vec();
     let version = match parts.version {
         Version::HTTP_10 => "HTTP/1",
         Version::HTTP_11 => "HTTP/1.1",
@@ -75,7 +75,10 @@ fn run_callback(
         environ.set_item(pyo3::intern!(py, "SERVER_PORT"), server.1)?;
         environ.set_item(pyo3::intern!(py, "REMOTE_ADDR"), client)?;
         environ.set_item(pyo3::intern!(py, "REQUEST_METHOD"), parts.method.as_str())?;
-        environ.set_item(pyo3::intern!(py, "PATH_INFO"), path)?;
+        environ.set_item(
+            pyo3::intern!(py, "PATH_INFO"),
+            PyBytes::new_bound(py, &path).call_method1(pyo3::intern!(py, "decode"), (pyo3::intern!(py, "latin1"),))?,
+        )?;
         environ.set_item(pyo3::intern!(py, "QUERY_STRING"), query_string)?;
         environ.set_item(pyo3::intern!(py, "wsgi.url_scheme"), scheme)?;
         environ.set_item(pyo3::intern!(py, "wsgi.input"), Py::new(py, WSGIBody::new(rt, body))?)?;

